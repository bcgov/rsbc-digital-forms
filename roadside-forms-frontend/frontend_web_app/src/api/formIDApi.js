import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const FormIDApi = {
  post: async function (data) {
    const headers = {
      ...await createRequestHeader(),
    };

    try {
      const response = await api
        .request({
          url: "/api/v1/forms",
          method: "POST",
          headers: { ...headers },
          data: JSON.stringify(data),
        });
      return response.data;      
    } catch (error) {
      console.log(error);
    }
    return null;
  },

  get: async function () {
    const headers = {
      ...await createRequestHeader(),
    };
    return await api
      .request({
        url: "/api/v1/forms",
        method: "GET",
        headers: { ...headers },
      })
      .then((response) => {
        return response.data;
      })
      .catch((error) => {
        return {
          status: error.status,
          data: error.response,
        };
      });
  },
  patch: async function (data) {
    const headers = {
      ...await createRequestHeader(),
    };
    return await api
      .request({
        url: "/api/v1/forms",
        method: "PATCH",
        headers: { ...headers },
        data: { ...data },
      })
      .then((response) => {
        return response.data;
      })
      .catch((error) => {
        return {
          status: error.status,
          data: error.response,
        };
      });
  },
};
