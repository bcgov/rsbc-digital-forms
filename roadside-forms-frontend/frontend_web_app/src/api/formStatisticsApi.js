import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const FormStatisticsApi = {
  getStatistics: async function () {
    const headers = {
      ...await createRequestHeader(),
    };
    return await api
      .request({
        url: "/api/v1/forms/statistics", // Adjust this to match your API endpoint
        method: "GET",
        headers: { ...headers },
      })
      .then((response) => {
        return {
          status: response.status,
          data: response.data,
        };
      })
      .catch((error) => {
        return {
          status: error.status,
          data: error.response,
        };
      });
  },
};