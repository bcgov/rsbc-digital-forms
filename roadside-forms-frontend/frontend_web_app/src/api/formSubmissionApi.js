import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const FormSubmissionApi = {

    post: async function (data) {
      const headers = {
        ...createRequestHeader(),
      };
      console.log(data)
        const response = await api.request({
          url: "/api/v1/event",
          method: "POST",
          headers: {...headers},
          data: {...data},
        })
        console.log(response)
        return response.json()
      },
};