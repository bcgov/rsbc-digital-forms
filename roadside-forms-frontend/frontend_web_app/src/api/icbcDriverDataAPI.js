import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const ICBCDriverDataApi = {
  get: async function (driver_licence_no) {
    const headers = {
      ...createRequestHeader(),
    };
    const response = await api.request({
      url: `/api/v1/icbc/drivers/${driver_licence_no}`,
      method: "GET",
      headers: { ...headers },
    });
    console.log(response);
    return response;
  },
};
