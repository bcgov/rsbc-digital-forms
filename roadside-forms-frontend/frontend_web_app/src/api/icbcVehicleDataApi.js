import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const ICBCVehicleDataApi = {
  get: async function (licencePlate) {
    const headers = {
      ...createRequestHeader(),
    };
    return await api
      .request({
        url: `/api/v1/icbc/vehicles/${licencePlate}`,
        method: "GET",
        headers: { ...headers },
      })
      .then((response) => {
        return response.data;
      })
      .catch((error) => {
        return {
          status: error.status,
          data: error.response,
        };
      });
  },
};
