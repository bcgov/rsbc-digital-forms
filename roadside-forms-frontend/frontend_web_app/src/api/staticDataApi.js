import { api } from "./config/axiosConfig";
import { createRequestHeader } from "../utils/requestHeaders";

export const StaticDataApi = {

      get: async function (resource) {
        const headers = {
          ...createRequestHeader(),
        };
        return await api.request({
          url: `/api/v1/static/${resource}`,
          method: "GET",
          headers:{...headers},
        }).then ( (response) => {
          return {
              status: response.status,
              data: response.data
          }
      }).catch((error) =>{
          // console.log(error)
          return {
              status: error.status,
              data: error.response
          }
      })
      },
};
