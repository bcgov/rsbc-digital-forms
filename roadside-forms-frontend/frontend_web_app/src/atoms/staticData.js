import {atom} from 'recoil'

export const vehicles = atom({
    key:"allVehicles",
    default: [],
  });

export const vehicleStyles = atom({
    key:"allVehicleStyles",
    default: [],
  });

  export const vehicleColours = atom({
    key:"allVehicleColours",
    default: [],
  });

export const provinces = atom({
    key:"allProvinces",
    default: [],
  });

export const permissions = atom({
    key:"allPermissions",
    default: [],
  });

export const jurisdictions = atom({
    key:"allJurisdictions",
    default: [],
  });

export const impoundLotOperators = atom({
    key:"allImpoundLotOperators",
    default: [],
  });

export const countries = atom({
    key:"allCountries",
    default: [],
  });

export const cities = atom({
    key:"allCities",
    default: [],
  });

export const agencies = atom({
    key:"allAgencies",
    default: [],
  });
