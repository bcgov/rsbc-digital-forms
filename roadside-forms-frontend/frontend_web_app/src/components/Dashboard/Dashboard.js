import React, { useEffect } from 'react';
import EditIcon from '@mui/icons-material/Edit';
import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import AddIcon from '@mui/icons-material/Add';
import Table from 'react-bootstrap/Table';
import { useSetRecoilState} from 'recoil';
import { staticResources } from '../../utils/helpers';
import { StaticDataApi } from '../../api/staticDataApi';
import { Button } from '../common/Button/Button';
import { useNavigate} from 'react-router-dom';
import './dashboard.scss'

export const Dashboard = () => {
  const navigate = useNavigate();
  const setAgencyResource = useSetRecoilState(staticResources["agencies"]);
  const setCityResource = useSetRecoilState(staticResources["cities"]);
  const setCountryResource = useSetRecoilState(staticResources["countries"]);
  const setJurisdictionResource = useSetRecoilState(staticResources["jurisdictions"]);
  const setImpoundResource = useSetRecoilState(staticResources["impound_lot_operators"]);
  const setProvinceResource = useSetRecoilState(staticResources["provinces"]);
  const setVehicleStyleResource = useSetRecoilState(staticResources["vehicle_styles"]);
  const setVehicleColourResource = useSetRecoilState(staticResources["vehicle_colours"]);
  const setVehicleResource = useSetRecoilState(staticResources["vehicles"]);

  useEffect(() => {
    const fetchData = async () => {
      try {
          const agencyData = await StaticDataApi.get("agencies")
          const cityData = await StaticDataApi.get("cities")
          const contryData = await StaticDataApi.get("countries")
          const jurisdictionData = await StaticDataApi.get("jurisdictions")
          const impoundData = await StaticDataApi.get("impound_lot_operators")
          const provinceData = await StaticDataApi.get("provinces")
          const vehicleStyleData = await StaticDataApi.get("vehicle_styles")
          const vehicleColourData = await StaticDataApi.get("vehicle_colours")
          const vehicleData = await StaticDataApi.get("vehicles")
          
          setVehicleResource(vehicleData.data)
          setVehicleStyleResource(vehicleStyleData.data)
          setVehicleColourResource(vehicleColourData.data)
          setProvinceResource(provinceData.data)
          setImpoundResource(impoundData.data)
          setJurisdictionResource(jurisdictionData.data)
          setCountryResource(contryData.data)
          setCityResource(cityData.data)
          setAgencyResource(agencyData.data)
        } catch (error) {
          console.error('Error fetching data:', error);
        }
      };
  
       fetchData();
    }, [
      setVehicleResource,
      setVehicleColourResource,
      setVehicleStyleResource,
      setProvinceResource,
      setImpoundResource,
      setJurisdictionResource,
      setCountryResource,
      setCityResource,
      setAgencyResource,
    ]);


  const handleClick = () => {
    navigate("/createEvent");
  }
      return (
      <>
      <div className='border-design text-font'>
        <div className='dashboard-header'>
          <h3><EditIcon/>Events in Progress</h3>
          <Button primary size="large" onClick={handleClick} label="New Event" icon={<AddIcon/>} />
        </div>
        <hr className='hr' />
        <Table>
          <thead>
            <tr>
              <th>Date & Time</th>
              <th>Form Types</th>
              <th>Location</th>
              <th>Surname</th>
              <th>Plate #</th>
              <th>Next Step</th>
            </tr>
          </thead>
        </Table>
      </div>
      <div className='border-design text-font'>
        <div className='dashboard-header'>
          <h3><ErrorOutlineIcon/>Waiting for Transmission to Server</h3>
          <span>Automatically re-trying in x seconds</span>
        </div>
        <hr className='hr' />
        <Table>
          <thead>
            <tr>
              <th>Date & Time</th>
              <th>Form #</th>
              <th>Form Type</th>
              <th>Location</th>
              <th>Surname</th>
              <th>Plate #</th>
            </tr>
          </thead>
        </Table>
        </div>
        <div className='border-design text-font'>
        <div className='dashboard-header'>
          <h3><CheckCircleOutlineIcon/>Completed</h3>
          <span>Last updated on date</span>
        </div>
        <hr className='hr' />
        <Table>
          <thead>
            <tr>
              <th>Date & Time</th>
              <th>Form #</th>
              <th>Form Type</th>
              <th>Location</th>
              <th>Surname</th>
              <th>Plate #</th>
            </tr>
          </thead>
        </Table>
      </div>
      </>
    );
  }
