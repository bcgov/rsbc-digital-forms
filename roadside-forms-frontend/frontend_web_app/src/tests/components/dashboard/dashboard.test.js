import React from 'react';
import { render, act } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { Dashboard } from '../../../components/Dashboard/Dashboard';
import { StaticDataApi } from '../../../api/staticDataApi';
import { vehicleStyles, jurisdictions, provinces, vehicles, vehicleColours, cities, impoundLotOperators } from '../../../atoms/staticData';
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
  }));
describe('Dashboard component', () => {
    test('renders dashboard component correctly', async () => {
    StaticDataApi.get = jest.fn().mockResolvedValue({});
    
    let container; 

    await act(async () => {
      ({ container } = await render( 
        <RecoilRoot
          initializeState={(snap) => {
            snap.set(vehicleStyles, [  {
                "code": "2DR",
                "name": "2-DOOR SEDAN"
            },
            {
                "code": "3DR",
                "name": "3-DOOR HATCH"
            }]);
            snap.set(jurisdictions, [{
                "objectCd": "AL",
                "objectDsc": "ALABAMA"
            },
            {
                "objectCd": "AK",
                "objectDsc": "ALASKA"
            }]);
            snap.set(provinces, [{
                "objectCd": "AK",
                "objectDsc": "ALASKA"
            },
            {
                "objectCd": "ALTA",
                "objectDsc": "ALBERTA"
            }]);
            snap.set(vehicles, [{
                "mk": "AC",
                "search": "A C (GREAT BRITAIN)",
                "md": ""
            },
            {
                "mk": "AC",
                "search": "A C (GREAT BRITAIN) - 3000 ME",
                "md": "300"
            }]);
            snap.set(vehicleColours, [
                {"code": 'test', 
                "display_name": 'test'}
            ]);
            snap.set(cities, [{
                "objectCd": "OH3H",
                "objectDsc": "100 TEST 1 HOUSE"
            },
            {
                "objectCd": "OHdM",
                "objectDsc": "106 TEST HOUSE"
            }]);
            snap.set(impoundLotOperators, [
            {
                "name": "Tset",
                "lot_address": "Test ST",
                "city": "TEST",
                "phone": "250-000-5343"
            },
            {
                "name": "Tset",
                "lot_address": "Test ST",
                "city": "TEST",
                "phone": "250-000-5343"
            }]);
          }}
        >
          <Dashboard />
        </RecoilRoot>
      ));
    });

    expect(container).toMatchSnapshot();
});
});
