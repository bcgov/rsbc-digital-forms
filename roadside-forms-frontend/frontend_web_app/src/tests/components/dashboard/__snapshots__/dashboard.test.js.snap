// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders dashboard component correctly 1`] = `
<div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        Events in Progress
      </h3>
      <button
        class="storybook-button storybook-button--large storybook-button--primary"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        New Event
      </button>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form Types
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
          <th>
            Next Step
          </th>
        </tr>
      </thead>
    </table>
  </div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        Waiting for Transmission to Server
      </h3>
      <span>
        Automatically re-trying in x seconds
      </span>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form #
          </th>
          <th>
            Form Type
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
        </tr>
      </thead>
    </table>
  </div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckCircleOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        Completed
      </h3>
      <span>
        Last updated on date
      </span>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form #
          </th>
          <th>
            Form Type
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;

exports[`renders dashboard component correctly 1`] = `
<div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        Events in Progress
      </h3>
      <button
        class="storybook-button storybook-button--large storybook-button--primary"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        New Event
      </button>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form Types
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
          <th>
            Next Step
          </th>
        </tr>
      </thead>
    </table>
  </div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        Waiting for Transmission to Server
      </h3>
      <span>
        Automatically re-trying in x seconds
      </span>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form #
          </th>
          <th>
            Form Type
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
        </tr>
      </thead>
    </table>
  </div>
  <div
    class="border-design text-font"
  >
    <div
      class="dashboard-header"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckCircleOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        Completed
      </h3>
      <span>
        Last updated on date
      </span>
    </div>
    <hr
      class="hr"
    />
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date & Time
          </th>
          <th>
            Form #
          </th>
          <th>
            Form Type
          </th>
          <th>
            Location
          </th>
          <th>
            Surname
          </th>
          <th>
            Plate #
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
